thirdparty/papertrail.go

```diff
@@ -0,0 +1,127 @@
+package thirdparty
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/pkg/errors"
+)
+
+type PapertrailSpan struct {
+	ID        string `json:"id"`
+	SHA256sum string `json:"sha256sum"`
+	Filename  string `json:"filename"`
+	Build     string `json:"build"`
+	Platform  string `json:"platform"`
+	Product   string `json:"product"`
+	Version   string `json:"version"`
+	Submitter string `json:"submitter"`
+	KeyID     string `json:"key_id"`
+	Time      uint64 `json:"time"`
+}
+
+type PapertrailProductVersion struct {
+	Product string           `json:"product"`
+	Version string           `json:"version"`
+	Spans   []PapertrailSpan `json:"spans"`
+}
+
+type PapertrailClient struct {
+	*http.Client
+	Address   string
+	KeyID     string
+	SecretKey string
+}
+
+type TraceArgs struct {
+	Build     string
+	Platform  string
+	Filename  string
+	Sha256    string
+	Product   string
+	Version   string
+	Submitter string
+}
+
+func (c *PapertrailClient) Trace(ctx context.Context, args TraceArgs) error {
```

Comment:
As public functions and structs, these should all have comments (since this code is maintained by you guys it maybe doesn't need to follow our best practices, but I think it would still be overall helpful, especially if it falls out of maintenance)

Reply:
I added comments to places that seemed reasonable!